src/ring_theory/hahn_series.lean

```diff
@@ -719,4 +733,250 @@ dif_neg hx
+(is_wf_Union_support' : set.is_wf (⋃ (a : α), (to_fun a).support))
+(co_support : Γ → finset α)
+(mem_co_support' : ∀ (a : α) (g : Γ), a ∈ co_support g ↔ (to_fun a).coeff g ≠ 0)
+
+end
+
+namespace summable_family
+section add_comm_monoid
+
+variables [linear_order Γ] [add_comm_monoid R] {α : Type*}
+
+instance : has_coe_to_fun (summable_family Γ R α) :=
+⟨λ _, (α → hahn_series Γ R), to_fun⟩
+
+lemma is_wf_Union_support (s : summable_family Γ R α) : set.is_wf (⋃ (a : α), (s a).support) :=
+s.is_wf_Union_support'
+
+@[simp]
+lemma mem_co_support {s : summable_family Γ R α} {a : α} {g : Γ} :
+  a ∈ s.co_support g ↔ (s a).coeff g ≠ 0 := mem_co_support' _ _ _
+
+@[ext]
+lemma ext {s t : summable_family Γ R α} (h : ∀ (a : α), s a = t a) : s = t :=
+begin
+  have hs := s.mem_co_support',
+  have ht := t.mem_co_support',
+  cases s,
+  cases t,
+  simp only,
+  split,
+  { ext,
+    exact congr (congr rfl (h _)) rfl },
+  { ext g a,
+    refine (hs a g).trans (iff.trans _ (ht a g).symm),
+    simp only [ne.def, not_iff_not],
+    split; intro h',
+    { exact eq.trans (congr (congr rfl (h _).symm) rfl) h' },
+    { exact eq.trans (congr (congr rfl (h _)) rfl) h' } }
+end
+
+instance : has_add (summable_family Γ R α) :=
+⟨λ x y, { to_fun := x + y,
+    is_wf_Union_support' := (x.is_wf_Union_support.union y.is_wf_Union_support).mono (begin
+      rw ← set.Union_union_distrib,
+      exact set.Union_subset_Union (λ a, support_add_subset)
+    end),
+    co_support := λ g, ((x.co_support g) ∪ (y.co_support g)).filter
+      (λ a, (x a).coeff g + (y a).coeff g ≠ 0),
+    mem_co_support' := λ a g, begin
+      simp only [mem_union, mem_filter, mem_co_support, and_iff_right_iff_imp,
+        pi.add_apply, ne.def, add_coeff'],
+      contrapose!,
+      rintro ⟨hx, hy⟩,
+      simp [hx, hy],
+    end }⟩
+
+instance : has_zero (summable_family Γ R α) :=
+⟨⟨0, by simp, λ _, ∅, by simp⟩⟩
+
+instance : inhabited (summable_family Γ R α) := ⟨0⟩
+
+@[simp]
+lemma add_apply {s t : summable_family Γ R α} {a : α} : (s + t) a = s a + t a := rfl
+
+@[simp]
+lemma zero_apply {a : α} : (0 : summable_family Γ R α) a = 0 := rfl
+
+instance : add_comm_monoid (summable_family Γ R α) :=
+{ add := (+),
+  zero := 0,
+  zero_add := λ s, by { ext, apply zero_add },
+  add_zero := λ s, by { ext, apply add_zero },
+  add_comm := λ s t, by { ext, apply add_comm },
+  add_assoc := λ r s t, by { ext, apply add_assoc } }
+
+end add_comm_monoid
+
+instance [linear_order Γ] [add_comm_group R] {α : Type*} :
+  add_comm_group (summable_family Γ R α) :=
+{ neg := λ s, { to_fun := λ a, - s a,
+    is_wf_Union_support' := by { simp_rw [support_neg], exact s.is_wf_Union_support' },
+    co_support := s.co_support,
+    mem_co_support' := by simp },
+  add_left_neg := λ a, by { ext, apply add_left_neg },
+  .. summable_family.add_comm_monoid }
+
+@[simp]
+lemma neg_apply [linear_order Γ] [add_comm_group R] {α : Type*}
+  {s t : summable_family Γ R α} {a : α} : (s - t) a = s a - t a := rfl
```

Comment:
This lemma is called `neg` but appears to be about `sub`. Can you ensure you have both?